### PR TITLE
Add default font size

### PR DESCRIPTION
**Pull request by GretaD**
Add default font size on the variables

### CODE REVIEW

**Comment by GretaD** [comment]
@skjnldsv should i do the same here: https://github.com/nextcloud/nextcloud-vue/blob/master/styleguide/assets/variables.css

**Comment by skjnldsv** [comment]
You also need to use this value on our scripts now :)

Like there for example
https://github.com/nextcloud/server/blob/e1a0b3f98fb28ec3d193f32a85dd77795baf88f7/core/css/styles.scss#L92

**Review by marcoambrosini — APPROVED** [review]
:tada:

**Review by marcoambrosini — COMMENTED** [review]
The only thing is I wouldn't call it `font-size` only as it creates a bit of confusion with the css rule. Maybe `default-font-size` or `main-font-size`?

**Comment by skjnldsv** [comment]
@ma12-co have you checked the code?
There is a conflict between your change using `.9375em;` and this variable.
Could your provide guidance to know what/where needs to be changed for the server styling to use this.
And why are we not using `px` but `em` :)
Thanks! :v:

**Review by skjnldsv — CHANGES_REQUESTED** [review]
Blocking to make sure we're also using the variable on the places where we have a font-size on server :)

Also see https://github.com/nextcloud/server/pull/23267

**Comment by ma12-co** [comment]
> Could your provide guidance to know what/where needs to be changed for the server styling to use this.

I think it would be enough to set:
`font-size: var(--root-font-size)` here:
https://github.com/nextcloud/server/blob/e1a0b3f98fb28ec3d193f32a85dd77795baf88f7/core/css/styles.scss#L92

No need to use `em` values anymore imo, we were using it just because the browser's default font size is 16px and we were just multiplying it with different coefficients. I see no advantages in that.
Also, since you're doing this @GretaD, could you make a variable for `line-height` as well and replace the line below with it too? It should be 24px.
`line-height: var(--root-line-height)`
https://github.com/nextcloud/server/blob/e1a0b3f98fb28ec3d193f32a85dd77795baf88f7/core/css/styles.scss#L93

**Comment by GretaD** [comment]
> Blocking to make sure we're also using the variable on the places where we have a font-size on server :)

you mean everywhere? Like where we have `font-size: 20px` -- i should do `font-size: $default-font-size +5 ` Or only where we use the default one.? I changed the ones you and Marco asked.

**Comment by skjnldsv** [comment]
> you mean everywhere? Like where we have `font-size: 20px` -- i should do `font-size: $default-font-size +5 ` Or only where we use the default one.? I changed the ones you and Marco asked.

No, only where we use 15 (might even be some leftovers 14 somewhere, feel free to adjust)

**Comment by GretaD** [comment]
> > you mean everywhere? Like where we have `font-size: 20px` -- i should do `font-size: $default-font-size +5 ` Or only where we use the default one.? I changed the ones you and Marco asked.
> 
> No, only where we use 15 (might even be some leftovers 14 somewhere, feel free to adjust)

good :) will update asap

**Comment by GretaD** [comment]
ready for review, i think i changed all the default ones

File: apps/settings/src/components/Markdown.vue [hunk 1]
```diff
@@ -146,7 +146,7 @@ export default {
 	}
 
 	h6 {
-		font-size: 14px;
+		font-size: $default-font-size;
```
Comment (skjnldsv) [inline apps/settings/src/components/Markdown.vue:149]:
```suggestion
		font-size: var(--default-font-size);
```

File: core/src/components/UnifiedSearch/SearchResult.vue [hunk 2]
```diff
@@ -250,7 +250,7 @@ $margin: 10px;
 	}
 	&-line-two {
 		opacity: .7;
-		font-size: 14px;
+		font-size: $default-font-size;
```
Comment (skjnldsv) [inline core/src/components/UnifiedSearch/SearchResult.vue:253]:
```suggestion
		font-size: var(--default-font-size);
```

File: core/src/views/UnifiedSearch.vue [hunk 3]
```diff
@@ -744,7 +744,7 @@ $input-padding: 6px;
 
 		::v-deep .empty-content__title {
 			font-weight: normal;
-            font-size: 14px;
+            font-size: $default-font-size;
```
Comment (skjnldsv) [inline core/src/views/UnifiedSearch.vue:747]:
```suggestion
            font-size: var(--default-font-size);
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]
You need to use the css variables not the scss ones :)

File: apps/settings/src/components/Markdown.vue (hunk 1 above)
Comment (GretaD) [reply apps/settings/src/components/Markdown.vue:149]: ah damn

**Review by skjnldsv — APPROVED** [review]

**Review by marcoambrosini — APPROVED** [review]

**Pull request merged**